<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>511</width>
    <height>345</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Магнитный симулятор</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>icon.png</normaloff>icon.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QWidget" name="widget_2" native="true">
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <property name="spacing">
        <number>5</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QWidget" name="viewpane" native="true">
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <property name="spacing">
           <number>0</number>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QSlider" name="scaler">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>300</number>
            </property>
            <property name="pageStep">
             <number>10</number>
            </property>
            <property name="value">
             <number>50</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="tickPosition">
             <enum>QSlider::TicksBelow</enum>
            </property>
            <property name="tickInterval">
             <number>10</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="MyGraphicsView" name="surface">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="layoutDirection">
             <enum>Qt::LeftToRight</enum>
            </property>
            <property name="backgroundBrush">
             <brush brushstyle="NoBrush">
              <color alpha="255">
               <red>0</red>
               <green>0</green>
               <blue>0</blue>
              </color>
             </brush>
            </property>
            <property name="foregroundBrush">
             <brush brushstyle="NoBrush">
              <color alpha="255">
               <red>0</red>
               <green>0</green>
               <blue>0</blue>
              </color>
             </brush>
            </property>
            <property name="interactive">
             <bool>false</bool>
            </property>
            <property name="sceneRect">
             <rectf>
              <x>0.000000000000000</x>
              <y>0.000000000000000</y>
              <width>0.000000000000000</width>
              <height>0.000000000000000</height>
             </rectf>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <property name="renderHints">
             <set>QPainter::Antialiasing|QPainter::SmoothPixmapTransform</set>
            </property>
            <property name="dragMode">
             <enum>QGraphicsView::ScrollHandDrag</enum>
            </property>
            <property name="transformationAnchor">
             <enum>QGraphicsView::AnchorUnderMouse</enum>
            </property>
            <property name="resizeAnchor">
             <enum>QGraphicsView::AnchorUnderMouse</enum>
            </property>
            <property name="viewportUpdateMode">
             <enum>QGraphicsView::MinimalViewportUpdate</enum>
            </property>
            <property name="optimizationFlags">
             <set>QGraphicsView::IndirectPainting</set>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="widget" native="true">
      <layout class="QHBoxLayout" name="horizontalLayout">
       <property name="spacing">
        <number>3</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QToolButton" name="refreshBtn">
         <property name="text">
          <string>...</string>
         </property>
         <property name="icon">
          <iconset theme="view-refresh" resource="icons.qrc">
           <normaloff>:/icons/refresh.ico</normaloff>:/icons/refresh.ico</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>16</width>
           <height>16</height>
          </size>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="dbgButton">
         <property name="text">
          <string>dbg</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="infoLbl">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="modeLbl">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Режим:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="modeBox">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="editable">
          <bool>false</bool>
         </property>
         <property name="currentText">
          <string>Просмотр</string>
         </property>
         <item>
          <property name="text">
           <string>Просмотр</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Поворот</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Поля (h)</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>511</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="fileMenu">
    <property name="title">
     <string>Файл</string>
    </property>
    <widget class="QMenu" name="examplesMenu">
     <property name="title">
      <string>Примеры</string>
     </property>
    </widget>
    <addaction name="openBtn"/>
    <addaction name="saveBtn"/>
    <addaction name="saveAsBtn"/>
    <addaction name="export_jpg"/>
    <addaction name="export_pdf"/>
    <addaction name="exitBtn"/>
    <addaction name="examplesMenu"/>
   </widget>
   <widget class="QMenu" name="viewMenu">
    <property name="title">
     <string>Вид</string>
    </property>
    <addaction name="particleProperties"/>
    <addaction name="systemProperties"/>
    <addaction name="doubleArrows"/>
    <addaction name="autoScale"/>
    <addaction name="ruler"/>
    <addaction name="refreshMenuItem"/>
   </widget>
   <widget class="QMenu" name="infoMenu">
    <property name="title">
     <string>Инфо</string>
    </property>
    <addaction name="aboutBtn"/>
   </widget>
   <addaction name="fileMenu"/>
   <addaction name="viewMenu"/>
   <addaction name="infoMenu"/>
  </widget>
  <action name="openBtn">
   <property name="icon">
    <iconset theme="document-open">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Открыть</string>
   </property>
   <property name="toolTip">
    <string>Открыть файл магнитной системы</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
   <property name="autoRepeat">
    <bool>false</bool>
   </property>
   <property name="menuRole">
    <enum>QAction::QuitRole</enum>
   </property>
  </action>
  <action name="saveBtn">
   <property name="icon">
    <iconset theme="document-save">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Сохранить</string>
   </property>
   <property name="toolTip">
    <string>Сохранить систему в файл</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="saveAsBtn">
   <property name="icon">
    <iconset theme="document-save-as">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Сохранить как</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="export_jpg">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Экспорт jpg</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Alt+J</string>
   </property>
  </action>
  <action name="export_pdf">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Экспорт pdf</string>
   </property>
   <property name="toolTip">
    <string>Экспортировать в векторный формат pdf</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Alt+P</string>
   </property>
  </action>
  <action name="particleProperties">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Свойства частицы</string>
   </property>
   <property name="toolTip">
    <string>Показать диалог с характеристиками выбранной частицы</string>
   </property>
  </action>
  <action name="systemProperties">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Свойства системы</string>
   </property>
   <property name="toolTip">
    <string>Показать диалог с параметрами системы</string>
   </property>
  </action>
  <action name="doubleArrows">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Стрелки двойной длины</string>
   </property>
   <property name="toolTip">
    <string>Если включено, стрелки проходят точку частицы в середине своей длины, иначе стрелки выходят из точки.</string>
   </property>
  </action>
  <action name="autoScale">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Авто масштаб</string>
   </property>
   <property name="toolTip">
    <string>Если включено, стрелки будут автоматически увеличены до нужного размера</string>
   </property>
  </action>
  <action name="ruler">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Линейка</string>
   </property>
   <property name="toolTip">
    <string>Показать линейку по периметру отображения</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="aboutBtn">
   <property name="icon">
    <iconset theme="dialog-information">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>О программе</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+I</string>
   </property>
  </action>
  <action name="exitBtn">
   <property name="icon">
    <iconset theme="application-exit">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Выход</string>
   </property>
   <property name="shortcut">
    <string>Alt+X</string>
   </property>
  </action>
  <action name="refreshMenuItem">
   <property name="icon">
    <iconset theme="view-refresh">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Обовить</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="actionAsd">
   <property name="text">
    <string>asd</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MyGraphicsView</class>
   <extends>QGraphicsView</extends>
   <header>mygraphicsview.h</header>
   <slots>
    <slot>changeOperateMode(int)</slot>
    <slot>dbgSlot()</slot>
    <slot>toggleDoubleArrows(bool)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>saveBtn</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveParticles()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>112</x>
     <y>24</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openBtn</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>loadParticles()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>216</x>
     <y>23</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>autoScale</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>toggleAutoCoff(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>255</x>
     <y>172</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveAsBtn</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveAsParticles()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>255</x>
     <y>172</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>modeBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>surface</receiver>
   <slot>changeOperateMode(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>456</x>
     <y>324</y>
    </hint>
    <hint type="destinationlabel">
     <x>255</x>
     <y>177</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>exitBtn</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>255</x>
     <y>172</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleArrows</sender>
   <signal>toggled(bool)</signal>
   <receiver>surface</receiver>
   <slot>toggleDoubleArrows(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>255</x>
     <y>181</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>refreshBtn</sender>
   <signal>clicked()</signal>
   <receiver>refreshMenuItem</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>323</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>refreshMenuItem</sender>
   <signal>triggered()</signal>
   <receiver>surface</receiver>
   <slot>update()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>255</x>
     <y>181</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dbgButton</sender>
   <signal>clicked()</signal>
   <receiver>surface</receiver>
   <slot>dbgSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>72</x>
     <y>324</y>
    </hint>
    <hint type="destinationlabel">
     <x>255</x>
     <y>181</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>saveParticles()</slot>
  <slot>loadParticles()</slot>
  <slot>clearParticles()</slot>
  <slot>toggleAutoCoff(bool)</slot>
  <slot>saveAsParticles()</slot>
 </slots>
</ui>
